### PR TITLE
Deploy playground manifests from repo

### PR DESCRIPTION
**Pull request by Pacman99**
The application is being synced here https://vela.scdev.aws.iohkdev.io/applications/marlowe-playground-repo/config. The status of all components can also be viewed when clicking on marlowe-staging at the top.

Some remaining TODOs that I don't believe are blockers:
 - [x] production and staging files have some duplication (not a whole lot since most env vars end up having different references anyways). ~The override policy method is kind of pointless here because the production and staging manifests have to be separate files anyway.~ Using the topology policy and two workflow steps they can now be in the same file
 - [ ] When a new image is pushed because of a code change, I'm not sure if flux will realize and re-deploy, because I think only manifest changes cause re-deploy. The optimal solution to this is to have flux/kubevela build images as part of the deploy workflow, but a temporary solution is to trigger the deploy manually after images are pushed.

Pre-submit checklist:
- Branch
    - [ ] Tests are provided (if possible)
    - [ ] Commit sequence broadly makes sense
    - [ ] Key commits have useful messages
    - [ ] Relevant tickets are mentioned in commit messages
    - [ ] Formatting, materialized Nix files, PNG optimization, etc. are updated
- PR
    - [ ] Self-reviewed the diff
    - [ ] Useful pull request description
    - [ ] Reviewer requested

### CODE REVIEW

**Comment by Pacman99** [comment]
status of application can be found here https://vela.scdev.aws.iohkdev.io/applications/marlowe-playground-marlowe-staging/config or here https://vela.scdev.aws.iohkdev.io/applications/marlowe-playground-repo/config. The latter is where the repository syncing is actually configured.

File: deploy/marlowe-playground.yaml [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+            valueFrom:
+              secretKeyRef:
+                key: clientSecret
+                name: gh-oauth-production
+          - name: GITHUB_CALLBACK_PATH
+            valueFrom:
+              secretKeyRef:
+                key: callbackPath
+                name: gh-oauth-production
+          image: ghcr.io/input-output-hk/marlowe-playground-server:2023-05-30-e2d5fd6
```
Comment (palas) [inline deploy/marlowe-playground.yaml]: Is this meant to be hardcoded?

Comment (palas) [reply deploy/marlowe-playground.yaml]: Ah, it is because it is production?

**Review by palas — APPROVED** [review]
Looks good. I imagine the thing would be to test it in practice, it is hard to know whether it will work otherwise

File: deploy/marlowe-playground.yaml (hunk 1 above)
Comment (Pacman99) [reply deploy/marlowe-playground.yaml]: Yes I was told there would be a production image, but thats not there yet so I just hardcoded it at the version Jamie deployed to production. Once the production image is available someone should change it.

**Pull request merged**